File: pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

char encryptForValuesInRegion(char value, char from, char to, int key);

int main(int argc, string argv[])
{
    if(argc != 2)
    {
        printf("Usage: %s k\n", argv[0]);
        return 1;
    }

    int keyDirty = atoi(argv[1]);
    int key = keyDirty % 26;

    printf("plaintext: ");
    string s = get_string();

    printf("ciphertext: ");
    for(int i = 0, n = strlen(s); i < n; i++)
    {
        char c = s[i];

        c = encryptForValuesInRegion(c, 'A', 'Z', key);
        c = encryptForValuesInRegion(c, 'a', 'z', key);

        printf("%c", c);
    }
    printf("\n");
}

char encryptForValuesInRegion(char value, char from, char to, int key)
{
    int cn = value;
    if(value >= from && value <= to)
    {
        cn = value + key;
        if(cn > to)
        {
            cn -= 26;
        }
    }
    return (char)cn;
}

File: pset7/caesar/caesar.py
import sys
from cs50 import get_string


def encryptForValuesInRegion(value, beginning, to, key):
    cn = ord(value)
    if value >= beginning and value <= to:
        cn += key
        if cn > ord(to):
            cn -= 26
    return chr(cn)


if len(sys.argv) != 2:
    print(f"Usage: {sys.argv[0]}\n")
    sys.exit(1)

keyDirty = int(sys.argv[1])
key = keyDirty % 26

s = get_string("plaintext: ")

print("ciphertext: ")
for i in range(len(s)):
    c = s[i]
    c = encryptForValuesInRegion(c, 'A', 'Z', key)
    c = encryptForValuesInRegion(c, 'a', 'z', key)

    print(c, end="")
print("")